client/client.go

```diff
@@ -0,0 +1,1375 @@
+	return SdkClientWriteRequest{
+		Client: *client,
+		ctx:    ctx,
+	}
+}
+
+func (request SdkClientWriteRequest) Options(options ClientWriteOptions) SdkClientWriteRequest {
+	request.options = &options
+	return request
+}
+
+func (request SdkClientWriteRequest) Body(body ClientWriteRequest) SdkClientWriteRequest {
+	request.body = &body
+	return request
+}
+
+func (request SdkClientWriteRequest) Execute() (ClientWriteResponse, error) {
+	return request.Client.WriteExecute(request)
+}
+
+func (client *OpenFgaClient) WriteExecute(request SdkClientWriteRequest) (ClientWriteResponse, error) {
+	var maxPerChunk = int32(1) // 1 has to be the default otherwise the chunks will be sent in transactions
+	if request.options != nil && request.options.Transaction != nil {
+		maxPerChunk = request.options.Transaction.MaxPerChunk
+	}
+	var maxParallelReqs = DEFAULT_MAX_METHOD_PARALLEL_REQS
+	if request.options != nil && request.options.Transaction != nil {
+		maxParallelReqs = request.options.Transaction.MaxParallelRequests
+	}
+
+	if request.options == nil || request.options.Transaction == nil || !request.options.Transaction.Disable {
+		writeRequest := openfga.WriteRequest{
+			AuthorizationModelId: client.getAuthorizationModelId(request.options.AuthorizationModelId),
+		}
+		if request.body.Writes != nil && len(*request.body.Writes) > 0 {
+			writes := openfga.TupleKeys{}
+			for index := 0; index < len(*request.body.Writes); index++ {
+				writes.TupleKeys = append(writes.TupleKeys, (*request.body.Writes)[index].ToTupleKey())
+			}
+			writeRequest.Writes = &writes
+		}
+		if request.body.Deletes != nil && len(*request.body.Deletes) > 0 {
+			deletes := openfga.TupleKeys{}
+			for index := 0; index < len(*request.body.Deletes); index++ {
+				deletes.TupleKeys = append(deletes.TupleKeys, (*request.body.Deletes)[index].ToTupleKey())
+			}
+			writeRequest.Deletes = &deletes
+		}
+		_, httpResponse, err := client.OpenFgaApi.Write(request.ctx).Body(writeRequest).Execute()
+
+		response := ClientWriteResponse{Writes: []ClientWriteSingleResponse{{}}, Deletes: []ClientWriteSingleResponse{{}}}
+
+		clientWriteStatus := SUCCESS
+		if err != nil {
+			clientWriteStatus = FAILURE
+		}
+
+		for index := 0; index < len(writeRequest.Writes.GetTupleKeys()); index++ {
+			response.Writes = append(response.Writes, ClientWriteSingleResponse{
+				TupleKey: (*request.body.Writes)[index],
+				Response: httpResponse,
+				Status:   clientWriteStatus,
+				Error:    err,
+			})
+		}
+		for index := 0; index < len(writeRequest.Deletes.GetTupleKeys()); index++ {
+			response.Deletes = append(response.Deletes, ClientWriteSingleResponse{
+				TupleKey: (*request.body.Deletes)[index],
+				Response: httpResponse,
+				Status:   clientWriteStatus,
+				Error:    err,
+			})
+		}
+		return response, err
+	}
+
+	var writeChunkSize = int(maxPerChunk)
+	var writeChunks [][]ClientTupleKey
+	for i := 0; i < len(*request.body.Writes); i += writeChunkSize {
+		end := i + writeChunkSize
```

Comment:
I am confused here

Reply:
(same as delete case)

Reply:
Will add some comments, but basically we are splitting the writes to chunks.

E.g. if we are passed: [a,b,c,d,e], and the `maxPerChunk` is `2`, we are generating: `[[a,b],[c,d],[e]]` so that later we can send three requests to the server

Reply:
I understand that...what I meant is that will it be better if we do something like 
`end=Min(i + writeChunkSize, len(*request.body.Writes))`

Note that math.Min may not be ideal as it is expect float. We can just write a simple utility to do min of integer